libs/auth/src/java_card.test.ts

```diff
@@ -0,0 +1,938 @@
+import { Buffer } from 'buffer';
+import * as fs from 'fs';
+import { sha256 } from 'js-sha256';
+import waitForExpect from 'wait-for-expect';
+import { assert } from '@votingworks/basics';
+import { electionFamousNames2021Fixtures } from '@votingworks/fixtures';
+import { mockOf } from '@votingworks/test-utils';
+import { Byte, UserWithCard } from '@votingworks/types';
+
+import {
+  getTestFilePath,
+  MockCardReader,
+  numericArray,
+  TestFileSetId,
+} from '../test/utils';
+import {
+  CardCommand,
+  constructTlv,
+  ResponseApduError,
+  SELECT,
+  STATUS_WORD,
+} from './apdu';
+import { Card, CardDetails, CheckPinResponse } from './card';
+import { CardReader } from './card_reader';
+import { DEV_JURISDICTION } from './certs';
+import {
+  CARD_VX_ADMIN_CERT,
+  CARD_VX_CERT,
+  DEFAULT_PIN,
+  GENERIC_STORAGE_SPACE,
+  GENERIC_STORAGE_SPACE_CAPACITY_BYTES,
+  JavaCard,
+  OPEN_FIPS_201_AID,
+  PUK,
+  VX_ADMIN_CERT_AUTHORITY_CERT,
+} from './java_card';
+import { DEV_PRIVATE_KEY_PASSWORD, JavaCardConfig } from './java_card_config';
+import {
+  certDerToPem,
+  createCert,
+  openssl,
+  PUBLIC_KEY_IN_DER_FORMAT_HEADER,
+  publicKeyDerToPem,
+} from './openssl';
+import {
+  construct8BytePinBuffer,
+  CRYPTOGRAPHIC_ALGORITHM_IDENTIFIER,
+  GENERAL_AUTHENTICATE,
+  GENERATE_ASYMMETRIC_KEY_PAIR,
+  GET_DATA,
+  PUT_DATA,
+  RESET_RETRY_COUNTER,
+  VERIFY,
+} from './piv';
+
+jest.mock('./card_reader');
+jest.mock('./openssl', (): typeof import('./openssl') => ({
+  // We use real openssl commands in these tests to ensure end-to-end correctness, the one
+  // exception being openssl commands for cert creation since two cert creation commands with the
+  // exact same inputs won't necessarily generate the same outputs, making assertions difficult
+  ...jest.requireActual('./openssl'),
+  createCert: jest.fn(),
+}));
```

Comment:
I'd initially planned to completely mock the `openssl` lib but then realized that at no point would I be testing the openssl commands underlying the Java Card implementation. With the new approach of using realistic test files and real openssl commands, I'm much more confident in our ability to prevent regressions